packages/firestore/src/local/local_store.ts

```diff
@@ -879,10 +879,11 @@ export class LocalStore {
     keepPersistedTargetData: boolean
   ): Promise<void> {
     const targetData = this.targetDataByTarget.get(targetId);
-    debugAssert(
-      targetData !== null,
-      `Tried to release nonexistent target: ${targetId}`
-    );
+
+    if (targetData === null) {
```

Comment:
How would this happen in practice that isn't a bug? I mean, how could we get in a state where the sync engine thought we were listening but local store didn't?

Reply:
I was trying to guard against this line: https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/core/sync_engine.ts#L1115

I did miss that this is only executed if the target is still active. I reverted this change.